terraform/stacks/agha_infra/main.tf

```diff
@@ -132,6 +139,67 @@ resource "aws_s3_bucket" "agha_gdr_store_2" {
     }
   }
 }
+
+resource "aws_s3_bucket_lifecycle_configuration" "agha_gdr_store_2" {
+  bucket = aws_s3_bucket.agha_gdr_store_2.id
+
+  # Will deep archive all 2022 MM data
+  rule {
+    status = "Enabled"
+    id = "[DeepArchive] MM_NSWHP/2022"
+
+    filter {
+      prefix = "MM_NSWHP/2022"
+    }
+
+    transition {
+      days = 0
+      storage_class = "DEEP_ARCHIVE"
+    }
+
+  }
+  rule {
+    status = "Enabled"
+    id = "[DeepArchive] MM_VCGS/2022"
+    
+    filter {
+      prefix = "MM_VCGS/2022"
+    }
+
+    transition {
+      days = 0
+      storage_class = "DEEP_ARCHIVE"
+    }
+  }
+
+  rule {
+    status = "Enabled"
+    id = "[DELETE NonCurrVer] MM_NSWHP/2022"
+    
+    filter {
+      prefix = "MM_NSWHP/2022"
+    }
+
+    noncurrent_version_expiration {
```

Comment:
Merge rules that apply to the same prefix? 
Or set this globally on the whole bucket (would we ever have prefix specific expiration settings?)

Reply:
Ahh right could do! Changing it, thanks.